src/models/ObservationPhoto.js

```diff
@@ -33,16 +32,33 @@ class ObservationPhoto extends Realm.Object {
   }
 
   static async new( uri ) {
-    const localFilePath = await resizeImageForUpload( uri );
-
     return {
+      _created_at: new Date( ),
+      _updated_at: new Date( ),
       uuid: uuid.v4( ),
-      photo: {
-        localFilePath
-      }
+      photo: await Photo.new( uri )
     };
   }
 
+  static async saveObservationPhoto( realm, photo ) {
+    const obsPhoto = await ObservationPhoto.new( photo.path );
+    realm?.write( ( ) => {
+      realm?.create( "ObservationPhoto", obsPhoto );
+    } );
+    return obsPhoto;
+  }
+
+  static async deleteObservationPhoto( realm, photo ) {
+    const photoToDelete = realm.objectForPrimaryKey( "ObservationPhoto", photo.uuid );
+
+    // delete localFilePath on disk
+    Photo.deleteLocalImage( photo.photo?.localFilePath );
+
+    realm?.write( ( ) => {
+      realm?.delete( photoToDelete );
```

Comment:
Note that this does not delete the Photo in Realm, since [Realm doesn't seem to support cascading deletes yet](https://github.com/realm/realm-swift/issues/1186). IMO, both the local file deletion and Photo deletion logic belong in the Photo model.

We're also not dealing with the fact that we can't really delete any of these things b/c eventually we need to tell the server about these deletions, but maybe we'll deal with that complexity later